File: kotlin-spring/src/main/kotlin/org/openapitools/model/RequestRoutesDepartureSearch.kt
package org.openapitools.model

import java.util.Objects
import com.fasterxml.jackson.annotation.JsonProperty
import org.openapitools.model.RequestRangeFull
import org.openapitools.model.RequestRoutesProperty
import org.openapitools.model.RequestTransportation
import javax.validation.constraints.DecimalMax
import javax.validation.constraints.DecimalMin
import javax.validation.constraints.Max
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull
import javax.validation.constraints.Pattern
import javax.validation.constraints.Size
import javax.validation.Valid

/**
 * 
 * @param id 
 * @param departureLocationId 
 * @param arrivalLocationIds 
 * @param transportation 
 * @param departureTime 
 * @param properties 
 * @param range 
 */
data class RequestRoutesDepartureSearch(

    @field:JsonProperty("id", required = true) val id: kotlin.String,

    @field:JsonProperty("departure_location_id", required = true) val departureLocationId: kotlin.String,

    @get:Size(min=1,max=2) 
    @field:JsonProperty("arrival_location_ids", required = true) val arrivalLocationIds: kotlin.collections.List<kotlin.String>,

    @field:Valid
    @field:JsonProperty("transportation", required = true) val transportation: RequestTransportation,

    @field:JsonProperty("departure_time", required = true) val departureTime: java.time.OffsetDateTime,

    @field:Valid
    @field:JsonProperty("properties", required = true) val properties: kotlin.collections.List<RequestRoutesProperty>,

    @field:Valid
    @field:JsonProperty("range") val range: RequestRangeFull? = null
) {

}

File: scala-akka-http-server/src/main/scala/org/openapitools/server/model/RequestRoutesDepartureSearch.scala
package org.openapitools.server.model

import java.time.OffsetDateTime

/**
 * @param id  for example: ''null''
 * @param departureLocationId  for example: ''null''
 * @param arrivalLocationIds  for example: ''null''
 * @param transportation  for example: ''null''
 * @param departureTime  for example: ''null''
 * @param properties  for example: ''null''
 * @param range  for example: ''null''
*/
final case class RequestRoutesDepartureSearch (
  id: String,
  departureLocationId: String,
  arrivalLocationIds: Seq[String],
  transportation: RequestTransportation,
  departureTime: OffsetDateTime,
  properties: Seq[RequestRoutesProperty],
  range: Option[RequestRangeFull]
)

